app/models/agents/gap_detector_agent.rb

```diff
@@ -8,6 +8,11 @@ class GapDetectorAgent < Agent
       The `value_path` value is a [JSONPath](http://goessner.net/articles/JsonPath/) to a value of interest. If either
       this value is empty, or no Events are received, during `window_duration_in_days`, an Event will be created with
       a payload of `message`.
+
+      The `stream_path` value is a [JSONPath](http://goessner.net/articles/JsonPath/) to a value in the event that will
```

Comment:
I am sorry that I did not mention my concern earlier, in the first PR I was only giving a general review of the code.

> They're not mutually exclusive.

As I see it currently they have kind of a dependency on each other, in `receive` `value_path` needs to be present, if `stream_path` is set as well it needs to have either the same path or be a descendant of `value_path`, only setting `stream_path` does not work. 
In `check` `stream_path` acts as a mode switch, if its present the values of the path are checked for gaps, when it's not set the old behavior is applied.

I might be missing something, but I don't see a use case where it would make sense to configure `value_path` and `stream_path`, that's why I suggested having one `value_path` option with a boolean/mode option that sets the behavior (check for gaps in the value presence or check for gaps in value stream).